include/istio/mixerclient/check_response.h

```diff
@@ -22,21 +22,15 @@
 namespace istio {
 namespace mixerclient {
 
-// The CheckResponseInfo holds response information in detail.
-struct CheckResponseInfo {
-  // Whether this check response is from cache.
-  bool is_check_cache_hit{false};
+// The CheckResponseInfo exposes policy and quota check details to the check
+// callbacks.
+class CheckResponseInfo {
+ public:
+  virtual ~CheckResponseInfo(){};
 
-  // Whether this quota response is from cache.
-  bool is_quota_cache_hit{false};
+  virtual const ::google::protobuf::util::Status& status() const = 0;
```

Comment:
Nit: Envoy uses `PURE` macro for `= 0`;

Reply:
Yeah, I think I'd have to define it here since we are in the mixerclient tree and no direct dependencies on Envoy are allowed.

Reply:
We discussed and decided that since istio/mixerclient can't depend on Envoy sources, it's ok to not make this change